provider/pkg/provider/deployment_schedules.go

```diff
@@ -0,0 +1,345 @@
+			v.Kind = v.Kind.AsReplace()
+			replaces = append(replaces, k)
+		}
+		detailedDiffs[k] = &pulumirpc.PropertyDiff{
+			Kind:      pulumirpc.PropertyDiff_Kind(v.Kind),
+			InputDiff: v.InputDiff,
+		}
+	}
+
+	changes := pulumirpc.DiffResponse_DIFF_NONE
+	if len(detailedDiffs) > 0 {
+		changes = pulumirpc.DiffResponse_DIFF_SOME
+	}
+	return &pulumirpc.DiffResponse{
+		Changes:             changes,
+		Replaces:            replaces,
+		DetailedDiff:        detailedDiffs,
+		HasDetailedDiff:     true,
+		DeleteBeforeReplace: len(replaces) > 0,
+	}, nil
+}
+
+func (st *PulumiServiceDeploymentScheduleResource) Delete(req *pulumirpc.DeleteRequest) (*pbempty.Empty, error) {
+	output, err := st.ToPulumiServiceDeploymentScheduleOutput(req.GetProperties())
+	if err != nil {
+		return nil, err
+	}
+
+	err = st.client.DeleteSchedule(context.Background(), output.Input.Stack, output.ScheduleID)
+	if err != nil {
+		return nil, err
+	}
+	return &pbempty.Empty{}, nil
+}
+
+func (st *PulumiServiceDeploymentScheduleResource) Create(req *pulumirpc.CreateRequest) (*pulumirpc.CreateResponse, error) {
+	input, err := st.ToPulumiServiceDeploymentScheduleInput(req.GetProperties())
+	if err != nil {
+		return nil, err
+	}
+
+	scheduleReq := pulumiapi.CreateDeploymentScheduleRequest{
+		ScheduleCron: input.ScheduleCron,
+		ScheduleOnce: input.ScheduleOnce,
+		Request: pulumiapi.CreateDeploymentRequest{
+			PulumiOperation: input.PulumiOperation,
+		},
+	}
+	scheduleID, err := st.client.CreateDeploymentSchedule(context.Background(), input.Stack, scheduleReq)
+	if err != nil {
+		return nil, err
+	}
+
+	outputProperties, err := plugin.MarshalProperties(
+		input.ToOutputPropertyMap(*scheduleID),
+		plugin.MarshalOptions{
+			KeepUnknowns: true,
+			SkipNulls:    true,
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return &pulumirpc.CreateResponse{
+		Id:         path.Join(input.Stack.OrgName, input.Stack.ProjectName, input.Stack.StackName, *scheduleID),
+		Properties: outputProperties,
+	}, nil
+}
+
+func (st *PulumiServiceDeploymentScheduleResource) Check(req *pulumirpc.CheckRequest) (*pulumirpc.CheckResponse, error) {
+	inputMap, err := plugin.UnmarshalProperties(req.GetNews(), plugin.MarshalOptions{KeepUnknowns: true, SkipNulls: true})
+	if err != nil {
+		return nil, err
+	}
+
+	var failures []*pulumirpc.CheckFailure
+	for _, p := range []resource.PropertyKey{"organization", "project", "stack", "pulumiOperation"} {
+		if !inputMap[(p)].HasValue() {
+			failures = append(failures, &pulumirpc.CheckFailure{
+				Reason:   fmt.Sprintf("missing required property '%s'", p),
+				Property: string(p),
+			})
+		}
+	}
+
+	if (inputMap["scheduleCron"].HasValue() && inputMap["timestamp"].HasValue()) ||
+		(!inputMap["scheduleCron"].HasValue() && !inputMap["timestamp"].HasValue()) {
+		failures = append(failures, &pulumirpc.CheckFailure{
+			Reason:   "One of scheduleCron and timestamp must be specified but not both",
+			Property: "projectName",
```

Comment:
This should be one of scheduleCron or timestamp - I think?

Reply:
Ah, agree "and" is confusing here, I'll change to "or"

Reply:
Again, was referring to the Property
```suggestion
			Property: "scheduleCron",
```

Reply:
Totally flew over my head, fixing